### DISCUSSION TITLE
Why would someone use Gravity over say, pihole or adguard?

### DISCUSSION THREAD

**Discussion by bonelifer** [BeryJu/gravity#1060] [Q&A]
Why would someone use Gravity over say, pi-hole or ad-guard?

**Comment by BeryJu** [comment]
The main reason why I started the project was to create a good, simple, all in one replacement for Active Directory DNS and DHCP, as I was really trying to get away from needing any windows in my infrastructure. Hence one of the big selling points of Gravity is the API for automation with terraform (which was always painful with AD DNS), and the inbuilt replication of records across all nodes, and having a nice web interface. Having ad-blocking came as more of an afterthought, and while noticing that Blocky uses the same DNS library as gravity does, so it was quite simple to integrate the two.

There aren't too many advantages if you have pihole/adguard and are happy with it, the main environments where gravity really shines is if you have multiple sites for DNS/DHCP and you want all the configuration to be replicated between the sites.
